src/app/home-page/components/slideshow/slideshow.component.ts

```diff
@@ -0,0 +1,205 @@
+    'http://forgedground.com/image/cache/catalog/viking-medieval-escudos-cascos-cuernos-espadass-accesorios-forgedground-1540x650.jpg',
+    'http://www.opencart.lionode.com/leoc04_2_2018/oc01/image/cache/catalog/banner%20main1-1540x650.jpg',
+    'http://www.opencart.lionode.com/leoc04_2_2018/oc01/image/cache/catalog/banner%20main2-1540x650.jpg'
+  ];
+
+  private slidesLength: number;
+  private translateStep: number;
+  private currentTranslatePosition: number;
+  private totalSlidesSize: number;
+  private firstItem: HTMLElement;
+  private lastItem: HTMLElement;
+  private counter = 0;
+  public selected = 0;
+  private intervalStart;
+  private subscriptions: Subscription[] = [];
+  public buttonsBackAndNext: HTMLElement[];
+  public sliderIsHovered = false;
+
+  constructor() {
+    this.moveToEdgeSlideWithoutRewind = this.moveToEdgeSlideWithoutRewind.bind(
+      this
+    );
+  }
+
+  ngOnInit(): void {
+    this.slidesLength = this.responseImgs.length;
+    this.totalSlidesSize = this.slidesLength * 100;
+    this.allSlides.nativeElement.style.width = this.totalSlidesSize + '%';
+    this.translateStep = 100 / this.slidesLength;
+    this.currentTranslatePosition = 0;
+    this.intervalStart = setInterval(() => {
+      this.moveSlide(SLIDE_DIRECTION.RIGHT);
+    }, 4000);
+  }
+
+  ngAfterViewInit(): void {
+    this.handleClick([
+      {
+        el: this.next.nativeElement,
+        callback: () => this.moveSlide(SLIDE_DIRECTION.RIGHT)
+      },
+      {
+        el: this.back.nativeElement,
+        callback: () => this.moveSlide(SLIDE_DIRECTION.LEFT)
+      }
+    ]);
+    this.firstItem = this.allSlides.nativeElement.querySelectorAll('.slide-item')[0];
+    this.lastItem = this.allSlides.nativeElement.querySelectorAll('.slide-item')[this.slidesLength - 1];
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(subscription => subscription.unsubscribe());
+    clearInterval(this.intervalStart);
+  }
+
+  private moveSlide(direction: SLIDE_DIRECTION): void {
+    this.allSlides.nativeElement.style.transition = '';
+    // default values, that matches LEFT direction to reduce code
+    let counter = 0;
+    let translatePostion = this.translateStep;
+    let step = this.currentTranslatePosition + this.translateStep;
+    let commonCounterAction = 'dec';
+    let itemToMove = this.lastItem;
+    let itemToMovePosition = -this.totalSlidesSize;
+    let counterAction = this.slidesLength - 1;
+
+    if (direction === SLIDE_DIRECTION.RIGHT) {
+      counter = this.slidesLength - 1;
+      translatePostion = -this.translateStep;
+      step = -100;
+      commonCounterAction = 'inc';
+      itemToMove = this.firstItem;
+      itemToMovePosition = this.totalSlidesSize;
+      counterAction = 0;
+    }
+
+    if (this.counter === counter) {
+      this.handleLastAndFirstMoves(step, itemToMove, itemToMovePosition, counterAction);
+    } else {
+      this.commonSlidesMove(translatePostion, commonCounterAction);
```

Comment:
try to be consistent
rename to moveEdgeSlides and moveSlides

Reply:
done